pydas/drivers.py

```diff
@@ -953,8 +953,10 @@ def generate_upload_token(self, token, item_id, filename, checksum=None):
         parameters['token'] = token
         parameters['itemid'] = item_id
         parameters['filename'] = filename
-        if checksum is not None:
-            parameters['checksum'] = checksum
+        optional_keys = ['checksum', 'create_additional_revision']
+        for key in optional_keys:
+            if key in kwargs:
+                parameters[key] = kwargs[key]
```

Comment:
Unless I am mistaken, if `checksum=None` or `create_additional_revision=None`, the parameter should not be passed to the request.